src/saturn_engine/worker/executors/arq/executor.py

```diff
@@ -81,15 +87,76 @@ def deserialize(self, data: bytes) -> dict[str, t.Any]:
         obj = pickle.loads(data)  # noqa: S301
         return obj
 
+    def create_connection_pool(self) -> ConnectionPool:
+        # Create a connection from
+        redis_settings = RedisSettings.from_dsn(self.options.redis_url)
+        kwargs: dict[str, t.Any] = {
+            "username": redis_settings.username,
+            "password": redis_settings.password,
+            "db": redis_settings.database,
+            "socket_connect_timeout": 1,
+            "max_connection": self.concurrency,
+            "timeout": 10,
+        }
+
+        if redis_settings.unix_socket_path is not None:
+            kwargs.update(
+                {
+                    "path": redis_settings.unix_socket_path,
+                    "connection_class": UnixDomainSocketConnection,
+                }
+            )
+        else:
+            # TCP specific options
+            kwargs.update(
+                {
+                    "host": redis_settings.host,
+                    "port": redis_settings.port,
+                }
+            )
+            if redis_settings.ssl:
+                kwargs["connection_class"] = SSLConnection
+
+        kwargs.update(self.options.redis_pool_args)
+        return BlockingConnectionPool(**kwargs)
+
     @cached_property
     async def redis_queue(self) -> ArqRedis:
-        redis_queue = await create_pool(
-            RedisSettings.from_dsn(self.options.redis_url),
-            job_serializer=self.serialize,
-            job_deserializer=self.deserialize,
-        )
-        await self.init_redis(redis_queue)
-        return redis_queue
+        def make_arq_redis() -> ArqRedis:
+            connection_pool = self.create_connection_pool()
+            return ArqRedis(
+                job_serializer=self.serialize,
+                job_deserializer=self.deserialize,
+                connection_pool=connection_pool,
+            )
+
+        retry = 0
+        while True:
```

Comment:
And this is taken from `arq` since it doesn't allo overriding the `connection_pool`